Automattic/wp-calypso
# Me: Create scaffold for me/domains page

Related to https://github.com/Automattic/martech/issues/1872

## Proposed Changes

* Adds the domains management all domains page to me/domains
* Note that if we were to implement this, then we'd be replacing the "Manage All Domains" link in the /me sidebar

## Screenshots
![2023-06-29 17 53 42](https://github.com/Automattic/wp-calypso/assets/5414230/14ab5495-86ce-4f56-9c64-d2fa59d78eb8)

## Testing Instructions



* Check out this branch
* `yarn && yarn start`
* Navigate to calypso.localhost:3000/me/domains

## Pre-merge Checklist
